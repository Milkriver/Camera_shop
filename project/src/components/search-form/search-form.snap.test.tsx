import { configureMockStore } from '@jedmao/redux-mock-store';
import {render} from '@testing-library/react';
import { createMemoryHistory } from 'history';
import HistoryRouter from '../history-router/history-router';
import thunk from 'redux-thunk';
import SearchForm from './search-form';
import { Provider } from 'react-redux';

const history = createMemoryHistory();

describe('Component: SearchForm', () => {
  it('should render correctly', () => {
    const middlewares = [thunk];
    const mockStore = configureMockStore(middlewares);

    const store = mockStore({
      DATA: {
        activePaginationPage: 1,
      },
      OFFERS: {
        searchedProducts: []
      }
    });
    const {container} = render(
      <Provider store={store}>
        <HistoryRouter history={history}>
          <SearchForm />
        </HistoryRouter>
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});
