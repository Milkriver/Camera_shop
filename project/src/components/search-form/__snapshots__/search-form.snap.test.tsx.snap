// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: SearchForm should render correctly 1`] = `
<div>
  <form>
    <label>
      <svg
        aria-hidden="true"
        class="form-search__icon"
        height="16"
        width="16"
      >
        <use
          xlink:href="#icon-lens"
        />
      </svg>
      <input
        autocomplete="off"
        class="form-search__input"
        placeholder="Поиск по сайту"
        type="text"
      />
    </label>
    <ul
      class="form-search__select-list"
    >
      <li
        class="form-search__select-item"
        tabindex="0"
      >
        Cannonball Pro MX 8i
      </li>
      <li
        class="form-search__select-item"
        tabindex="0"
      >
        Cannonball Pro MX 7i
      </li>
      <li
        class="form-search__select-item"
        tabindex="0"
      >
        Cannonball Pro MX 6i
      </li>
      <li
        class="form-search__select-item"
        tabindex="0"
      >
        Cannonball Pro MX 5i
      </li>
      <li
        class="form-search__select-item"
        tabindex="0"
      >
        Cannonball Pro MX 4i
      </li>
    </ul>
  </form>
</div>
`;
